### ISSUE TITLE
make utilities work with non-UTF-8 inputs

### ISSUE THREAD

**Issue by kwantam** [uutils/coreutils#554]
Since the `String` types in Rust are by nature UTF-8, many of these utilities make the implicit assumption that all inputs will be UTF-8 encoded. It might be useful to expand the capabilities to include non-UTF-8 string handling as well. #553 does this for `expand` and `unexpand`.

**Comment by remram44** [comment]
The problem is that getopts returns `String`s, so unless it gets patched or this project switches to something else, it's never going to work.

See also #203

**Comment by remram44** [comment]
@Arcterus the bot seems misconfigured, it's marking accepted issues as stale.

**Comment by sylvestre** [comment]
I configured the bot this way.
It didn't see any chance for 3 years, this is why it is marked as staled (which is clearly the case here)

**Comment by remram44** [comment]
For that definition, it is indeed stale, however it is still an open issue. Marking actual issues as "closed" hurts the community. Why do this?

**Comment by sylvestre** [comment]
Because keeping bugs open for years without any activity isn't providing any benefit for anyone and reopening bugs is cheap.

**Comment by remram44** [comment]
Reopening bugs is cheap for you, in that it annoys somebody else. I'm sorry you want to treat your community this way, but you can do what you want.

Luckily the real coreutils treat product quality better. If you're happy being a toy reimplementation rather than providing a usefuly rewrite, that is a valid choice. You should probably update your README though.

**Comment by Arcterus** [comment]
@remram44 I think you're going a bit too far there.

@sylvestre I'm not sure why you would want to close an issue for a bug just because there hasn't been activity.  It's not like the bug has disappeared.  If anything, I think this just makes it harder to keep track of things.  It's fine to mark stuff as stale, but only stale PRs should be closed IMO.

Also, I don't actually think this is stale.  All the PRs switching from `getopts` to `clap` help with this issue.

**Comment by sylvestre** [comment]
> Also, I don't actually think this is stale. All the PRs switching from getopts to clap help with this issue.

Yeah, you probably saw that I opened plenty of good first bugs for this. ;)
It is probably more concrete than 6 years old inactive bugs.

**Comment by Arcterus** [comment]
Well, this bug encompasses more than just switching to `clap` (although admittedly that is a significant portion of it).  So, I guess we can either split this into a new bug for every utility or make this more of tracking issue (the latter is my preference).

**Comment by sylvestre** [comment]
yeah, we fixed most of them
